### PR TITLE
KAFKA-15741: KRaft support in DescribeConsumerGroupTest

### PR DESCRIPTION
**Pull request by linzihao1999**
The tests in DescribeConsumerGroupTest in core/src/test/scala/unit/kafka/admin/DescribeConsumerGroupTest.scala update to support KRaft.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by linzihao1999** [comment]
@dengziming PTAL

**Comment by dengziming** [comment]
There seems some flaky related to this PR here: https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-14668/2/#showFailuresLink

**Comment by linzihao1999** [comment]
@dengziming Hi, I re-trigger the CI build 5 times and no flaky test about it.
https://ci-builds.apache.org/blue/organizations/jenkins/Kafka%2Fkafka-pr/detail/PR-14668/10/pipeline
https://ci-builds.apache.org/blue/organizations/jenkins/Kafka%2Fkafka-pr/detail/PR-14668/9/pipeline
https://ci-builds.apache.org/blue/organizations/jenkins/Kafka%2Fkafka-pr/detail/PR-14668/8/pipeline
https://ci-builds.apache.org/blue/organizations/jenkins/Kafka%2Fkafka-pr/detail/PR-14668/7/pipeline
https://ci-builds.apache.org/blue/organizations/jenkins/Kafka%2Fkafka-pr/detail/PR-14668/6/pipeline

File: core/src/test/scala/unit/kafka/admin/DescribeConsumerGroupTest.scala [hunk 1]
```diff
@@ -35,9 +35,11 @@ class DescribeConsumerGroupTest extends ConsumerGroupCommandTest {
   private val describeTypeState = Array(Array("--state"))
   private val describeTypes = describeTypeOffsets ++ describeTypeMembers ++ describeTypeState
 
-  @Test
-  def testDescribeNonExistingGroup(): Unit = {
-    TestUtils.createOffsetsTopic(zkClient, servers)
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumName)
+  @ValueSource(strings = Array("zk", "kraft"))
+  def testDescribeNonExistingGroup(quorum: String): Unit = {
+    createOffsetsTopic();
```
Comment (mimaison) [inline core/src/test/scala/unit/kafka/admin/DescribeConsumerGroupTest.scala]: Nit: Can we remove the unneeded semicolon?
Same below

File: core/src/test/scala/unit/kafka/admin/DescribeConsumerGroupTest.scala [hunk 2]
```diff
@@ -351,7 +367,9 @@ class DescribeConsumerGroupTest extends ConsumerGroupCommandTest {
     val (result, succeeded) = TestUtils.computeUntilTrue(service.collectGroupOffsets(group)) {
       case (state, assignments) =>
         val testGroupAssignments = assignments.toSeq.flatMap(_.filter(_.group == group))
+
```
Comment (mimaison) [inline core/src/test/scala/unit/kafka/admin/DescribeConsumerGroupTest.scala]: Let's try to not undo unnecessary changes like these

**Review by mimaison — COMMENTED** [review]
Thanks for the PR, I left a few comments

**Review by mimaison — APPROVED** [review]
LGTM, thanks!
